### PR TITLE
Clarify when to use TORCH_FORCE_NO_WEIGHTS_ONLY_LOAD

### PR DESCRIPTION
**Pull request by chtruong814**
# What does this PR do ?

Clarify when to use TORCH_FORCE_NO_WEIGHTS_ONLY_LOAD

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -289,6 +289,8 @@ multi-GPU/multi-node training.
 - Pytorch 2.6 or above
 - NVIDIA GPU (if you intend to do model training)
 
+As of Pytorch 2.6, `torch.load` defaults to using `weights_only=True`. Some model checkpoints may require using `weights_only=False`. In this case, you can set the env var `TORCH_FORCE_NO_WEIGHTS_ONLY_LOAD=1` before running code that uses `torch.load`. However, this should only be done with trusted files. Loading files from untrusted sources with more than weights only can have the risk of arbitrary code execution.
+
```
Comment (XuesongYang) [inline README.md:293]: LGTM. We could also link https://docs.pytorch.org/docs/stable/notes/serialization.html#torch-load-with-weights-only-true for reference

Comment (chtruong814) [reply README.md:293]: Good idea

Comment (pzelasko) [reply README.md:293]: Looks good, we should have an FAQ section maybe..
